### PR TITLE
[Enhancement] Support RotatedRetinaNet TensorRT

### PR DESCRIPTION
**Pull request by grimoire**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily receiving feedbacks. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

Add rotated nms TensorRT support.

## Modification

Refactor nms and rotated nms TensorRT code. Support Rotated Retinanet.
Benchmark is preparing.

This PR is required:
- [x] #428 

<table>
<tbody>
  <tr>
    <td align="center">Model</td>
    <td align="center">Task</td>
    <td align="center">Dataset</td>
    <td align="center">Metrics</td>
    <td align="center">fp32</td>
    <td align="center">fp32</td>
    <td align="center">fp32</td>
    <td align="center">fp16</td>
    <td align="center">fp16</td>
    <td align="center">fp32</td>
    <td>model config file</td>
  </tr>
  <tr>
    <td align="center" rowspan="2">RotatedRetinaNet</td>
    <td align="center" rowspan="2">Rotated Detection</td>
    <td align="center" rowspan="2">DOTA-v1.0</td>
    <td align="center">mAP</td>
    <td align="center">0.698</td>
    <td align="center">0.698</td>
    <td align="center">0.698</td>
    <td align="center">0.697</td>
    <td align="center">-</td>
    <td align="center">-</td>
    <td rowspan="2">$MMROTATE_DIR/configs/rotated_retinanet/rotated_retinanet_obb_r50_fpn_1x_dota_le135.py</td>
  </tr>
</tbody>
</table>

## BC-breaking (Optional)

Does the modification introduce changes that break the backward-compatibility of the downstream repositories?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here, and update the documentation.

## Checklist

1. Pre-commit or other linting tools are used to fix the potential lint issues.
2. The modification is covered by complete unit tests. If not, please add more unit tests to ensure the correctness.
3. If the modification has a dependency on downstream projects of a newer version, this PR should be tested with all supported versions of downstream projects.
4. The documentation has been modified accordingly, like docstring or example tutorials.

### CODE REVIEW

File: mmdeploy/mmcv/ops/nms_rotated.py [hunk 1]
```diff
@@ -74,3 +74,76 @@ def symbolic(g, boxes: Tensor, scores: Tensor, iou_threshold: float,
             scores,
             iou_threshold_f=float(iou_threshold),
             score_threshold_f=float(score_threshold))
+
```
Comment (RunningLeon) [inline mmdeploy/mmcv/ops/nms_rotated.py:77]: may add test for trt_rotated_nms in `tests/test_ops/test_ops.py`

File: csrc/backend_ops/tensorrt/common/nms/kernel.h [hunk 2]
```diff
@@ -47,6 +36,20 @@ pluginStatus_t allClassNMS(cudaStream_t stream, int num, int num_classes, int nu
...
                            void* afterNMS_scores, void* afterNMS_index_array, bool flipXY = false);
 
+pluginStatus_t allClassRotatedNMS(cudaStream_t stream, int num, int num_classes,
+                                  int num_preds_per_class, int top_k, float nms_threshold,
+                                  bool share_location, bool isNormalized, DataType DT_SCORE,
+                                  DataType DT_BBOX, void* bbox_data, void* beforeNMS_scores,
+                                  void* beforeNMS_index_array, void* afterNMS_scores,
+                                  void* afterNMS_index_array, bool flipXY = false);
+
+pluginStatus_t rotatedAllClassNMS(cudaStream_t stream, int num, int num_classes,
```
Comment (lzhangzz) [inline csrc/backend_ops/tensorrt/common/nms/kernel.h:46]: Why there are both `allClassRotatedNMS` and `rotatedAllClassNMS`?

File: tests/test_ops/test_ops.py [hunk 3]
```diff
@@ -379,6 +379,85 @@ def wrapped_function(boxes, scores):
...
+                             pre_topk,
+                             after_topk,
+                             iou_threshold,
+                             score_threshold,
+                             background_label_id,
+                             input_list=None,
+                             save_dir=None):
+    backend.check_env()
+    import importlib
+    if importlib.util.find_spec('mmrotate') is None:
```
Comment (RunningLeon) [inline tests/test_ops/test_ops.py:397]: could use `pytest.importorskip('mmrotate', reason='mmrorate is not installed.')`

File: docs/en/benchmark.md [hunk 4]
```diff
@@ -1855,8 +1855,8 @@ Users can directly test the performance through [how_to_evaluate_a_model.md](tut
     <td align="center">mAP</td>
     <td align="center">0.698</td>
     <td align="center">0.698</td>
-    <td align="center">-</td>
-    <td align="center">-</td>
+    <td align="center">0.698</td>
```
Comment (RunningLeon) [inline docs/en/benchmark.md:1858]: may update benchmark in zh-cn, supported_models.md, and mmrotate.md

**Review by RunningLeon — APPROVED** [review]
LGTM

**Review by irexyc — APPROVED** [review]

File: csrc/backend_ops/tensorrt/batched_rotated_nms/trt_batched_rotated_nms.cpp [hunk 5]
```diff
@@ -0,0 +1,238 @@
+// Copyright (c) OpenMMLab. All rights reserved
```
Comment (RunningLeon) [inline csrc/backend_ops/tensorrt/batched_rotated_nms/trt_batched_rotated_nms.cpp:1]: may  update the doc for trt ops: https://github.com/open-mmlab/mmdeploy/blob/master/docs/en/ops/tensorrt.md

**Review by RunningLeon — APPROVED** [review]
LGTM

**Review by irexyc — APPROVED** [review]

**Pull request merged**